### DISCUSSION TITLE
Framer animation has strange behaviour with chakra component

### DISCUSSION THREAD

**Discussion by skorphil** [chakra-ui/chakra-ui#8283] [Q&A] [answered]
Hi, i need to animate `Tab` component when layout is changing.

pure `motion.div` works as expected:

https://github.com/chakra-ui/chakra-ui/assets/6762009/1275f842-0281-4cc5-acf4-eb068c2e34aa

```jsx
<TabList>
    <motion.div
          key={id}
          layout="position"
          transition={{ duration: 0.2 }}
    >
...
```

But when i use chakra `Tab` with `as` prop i got strange behaviour:

https://github.com/chakra-ui/chakra-ui/assets/6762009/75cb0ccd-cc02-4e6a-8878-678baf627a4f

```jsx
<TabList>
    <InstitutionTab
        transition="2s linear"
        as={motion.div}
        layout="position"
        width={isKeyboardOpened && "180px"}
        key={`institutionTab-${id}`}
        name={institution.name}
        state="new"
    />
...
```

Do you know how to avoid this?

**Comment by skorphil** [comment]
Wrapping actually works for animation, but creates another problems:
```jsx
<motion.div
              key={id}
              layout="position"
              transition={{ duration: 0.2 }}
            >
              {/* {institution.name} */}
              <InstitutionTab
                width={isKeyboardOpened && "180px"}
                key={`institutionTab-${id}`}
                name={institution.name}
                state="new"
              />
            </motion.div>
```

**Comment by segunadebayo** [comment, marked as answer]
v3 dropped `framer-motion`. `as={motion.div}` on a Tab is a v2 pattern.

Animate with CSS (`_open` / `_closed`) or wrap the tab in a motion element yourself. We are not shipping new v2 work.
